<a href="https://colab.research.google.com/github/sovikapoudel/Data-Visualiation-Spring-2022/blob/main/Quiz_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1><center>Data Visualization Spring 2022</center></h1>

---

###Assignment: 5
Name: Sovika Poudel <br>
Year: Junior <br>
Course: CPSMA-4313 Data Processing Visualization 
### For this part of Assignment I am going to load the libraries, gather dataset and load them here in this notebook.

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Gathering Data

In [20]:
%%bigquery --project data-vis-assignment-338801
SELECT *
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  lIMIT 5



trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19

[5 rows x 9 columns]

What was the second most popular starting station?

In [11]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, COUNT(start_station_id) as total_starting_point
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  GROUP BY start_station_name
  Order BY total_starting_point DESC ###this will give the most popular on the top so we can see 2nd most popular in 2nd row
  LIMIT 5

start_station_name  total_starting_point
0      21st & Speedway @PCL                 72799
1      Riverside @ S. Lamar                 40635
2  City Hall / Lavaca & 2nd                 36520
3            2nd & Congress                 35307
4      Rainey St @ Cummings                 34758

So the second most popular starting station is Riverside @ S. Lamar	with 40635 counts

How many trips lasted over an hour and were a round trip (started and stopped at the same station)?

At first, for this part I am just showing the list in order and confirming if the query is running correctly.

In [46]:
%%bigquery --project data-vis-assignment-338801 
SELECT start_station_name, end_station_name, duration_minutes as time_travelled
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  LIMIT 100

start_station_name  ... time_travelled
0                   Toomey Rd @ South Lamar  ...             82
1                          Waller & 6th St.  ...             76
2                          Waller & 6th St.  ...             75
3                          Waller & 6th St.  ...             74
4                          Waller & 6th St.  ...             73
..                                      ...  ...            ...
95                  Toomey Rd @ South Lamar  ...             88
96                  Toomey Rd @ South Lamar  ...             73
97                  Toomey Rd @ South Lamar  ...             72
98                         Waller & 6th St.  ...             61
99  Capital Metro HQ - East 5th at Broadway  ...            192

[100 rows x 3 columns]

Now, for the total numbers of loop that were over 60 minutes I am using a clause within a 'WHERE' clause

In [50]:
%%bigquery --project data-vis-assignment-338801 
SELECT COUNT(*) as No_of_Overthehour_loops
  From `bigquery-public-data.austin_bikeshare.bikeshare_trips`
  WHERE  start_station_name = end_station_name and  duration_minutes >60
  
  # Order by time_travelled DESC ### this line of code is not working correctly on this part. so commenting this out for now.
 

No_of_Overthehour_loops
0                    54795